app/models/policy.rb

```diff
@@ -50,12 +50,25 @@ def update_hosts(new_host_ids)
     return unless new_host_ids
 
     removed = policy_hosts.where.not(host_id: new_host_ids).destroy_all
-    imported = PolicyHost.import((new_host_ids - host_ids).map do |host_id|
-      { host_id: host_id, policy_id: id }
-    end)
+    imported = PolicyHost.import_from_policy(id, new_host_ids - host_ids)
+    update_os_versions
+
     [imported.ids.count, removed.count]
   end
 
+  def update_os_versions
+    os_versions = Host.os_versions(hosts.pluck(:id))
+    initial_profile.update(os_version: os_versions.pop) if os_versions.any?
+
+    os_versions.each do |os_version|
+      initial_profile.parent_profile.clone_to(
```

Comment:
I'm not sure if this is correct to clone by the initial profile's parent. Each other profile would have a different parent since it is from a different benchmark. You can check the report parsing, where the clone to is used from/with a profile that is generated from the report.

Reply:
Agreed. @skateman and I have already discussed this.